4/mdav2/mp2.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+import psi4
+import sys
+sys.path.append('../../3/mdav2')
+import rhf
+
+def mp2(C, g, e, nocc):
+#Calculates mp2 (moller-plesset @ 2nd order) correlation energy.
+#Takes in a coefficient matrix (C), two-electron integrals (g),
+#orbital energies (e), and an occupation number (nocc)
+    phi = np.einsum('pQRS,pP->PQRS',
```

Comment:
Einsum allows you to multiply and sum over several arrays so you could do this calculation with just one line: "phi = np.einsum('pqrs,pP,qQ,rR,sS->PQRS', g, C, C, C, C)"